### PR TITLE
added the Action DELETE_MANAGED_URL

### PR DESCRIPTION
**Pull request by tylerslijboom-r7**
TOR-MBP-3824:mimecast tslijboom$ ./run.sh -R tests/get_managed_url.json -j
Running: cat tests/get_managed_url.json | docker run --rm   -i rapid7/mimecast:3.1.0  run | grep -- ^\{ | jq -r '.body | try(.log | split("\n") | .[]),.output'
rapid7/Mimecast:3.1.0. Step name: get_managed_url
Starting new HTTPS connection (1): sandbox-api.mimecast.com:443
https://sandbox-api.mimecast.com:443 "POST /api/ttp/url/get-all-managed-urls HTTP/1.1" 200 1357
rapid7/Mimecast:3.1.0. Step name: get_managed_url

{
  "response": [
    {
      "id": "wOi3MCwjYFYhZfkYlp2RMAhvN30QSmqOT7D-I9Abwlmy7ZH7eCwvY3ImP7QVjTLhHMy6V8J3VOvTNMW2G-txx3o4zL0YXqWxuCVlGQ-1viE",
      "scheme": "http",
      "domain": "youtube.com",
      "port": -1,
      "matchType": "explicit",
      "action": "block",
      "comment": "test",
      "disableUserAwareness": false,
      "disableRewrite": false,
      "disableLogClick": false
    }
  ]
}

TOR-MBP-3824:mimecast tslijboom$ icon-plugin run -A -R tests/delete_managed_url.json 
INFO[0000] Running command:  docker run --rm -i rapid7/mimecast:3.1.0 run < tests/delete_managed_url.json 
{"body": {"log": "rapid7/Mimecast:3.1.0. Step name: delete_managed_url\n", "status": "ok", "meta": {}, "output": {"success": true}}, "version": "v1", "type": "action_event"}

### Run

<details>

```
{
  "body": {
    "log": "rapid7/Mimecast:3.1.0. Step name: delete_managed_url\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/mimecast:3.1.0 run < tests/delete_managed_url.json
</summary>
</details>

### UI

Screenshots of the plugin being used in Komand for validating use go below this line.

#### Artifact

TOR-MBP-3824:mimecast tslijboom$ ./run.sh -R tests/get_managed_url.json -j
Running: cat tests/get_managed_url.json | docker run --rm   -i rapid7/mimecast:3.1.0  run | grep -- ^\{ | jq -r '.body | try(.log | split("\n") | .[]),.output'
rapid7/Mimecast:3.1.0. Step name: get_managed_url
Starting new HTTPS connection (1): sandbox-api.mimecast.com:443
https://sandbox-api.mimecast.com:443 "POST /api/ttp/url/get-all-managed-urls HTTP/1.1" 200 1310
rapid7/Mimecast:3.1.0. Step name: get_managed_url

{
  "response": []
}

### CODE REVIEW

File: mimecast/help.md [hunk 1]
```diff
@@ -119,6 +144,34 @@ Example output:
...
 
+#### Delete Managed URL
+
+This action is used to remove a Managed URL from the blocked list.
+
+##### Input
+
+|Name|Type|Default|Required|Description|Enum|
+|----|----|-------|--------|-----------|----|
+|id|string|None|True|Mimecast ID of the Managed URL|None|
```
Comment (lmilby-r7) [inline mimecast/help.md]:
```suggestion
|id|string|None|True|Mimecast ID of the managed URL|None|
```

File: mimecast/komand_mimecast/actions/delete_managed_url/action.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        id_to_remove = params.get('id')
+        if id_to_remove:
+            # Mimecast request, only if there is an ID to remove
+            mimecast_request = util.MimecastRequests()
+            response = mimecast_request.mimecast_post(url=url, uri=DeleteManagedUrl._URI,
+                                                      access_key=access_key, secret_key=secret_key,
+                                                      app_id=app_id, app_key=app_key, data={'id': id_to_remove})
+
+
+        return {'success': bool(1)}
```
Comment (lmilby-r7) [inline mimecast/komand_mimecast/actions/delete_managed_url/action.py]:
```suggestion
            return {'success': True}
```
Also If there is now id to remove this action will always return True. Move the `return` in to the check for `id`

File: mimecast/plugin.spec.yaml [hunk 3]
```diff
@@ -284,6 +284,12 @@ types:
       description: The action requested by the user
       type: string
       required: false
+  deletions:
+    deletedRecordIds:
+      title: Mimecast IDs of Deletions
+      description: The IDs of records deleted from the blocked list
+      type: array
```
Comment (lmilby-r7) [inline mimecast/plugin.spec.yaml]:
```suggestion
      type: array
```
`array` is not a type unless it was created in the custom type section. Please look at https://komand.github.io/python/spec.html?highlight=type#base-types and update the type for `deletedRecordIds`.

Comment (lmilby-r7) [reply mimecast/plugin.spec.yaml]: If the ids are strings just update to `"[]string"`

File: mimecast/help.md [hunk 4]
```diff
@@ -1,5 +1,30 @@
...
+## Setup
+
+No special instructions for setting up this plugin
+
+## Technical Details
+
+No notable technical requirements to highlight
+
+## About
+
```
Comment (jherlitz-r7) [inline mimecast/help.md]: Help markdown is not properly formatted here - it looks like there are some duplicate headings. 
Top level headings should be:
Description
Key Features
Requirements
Documentation
Version History
Links

Comment (jherlitz-r7) [reply mimecast/help.md]: it looks like you have appended some stuff to the top of the markdown that should not be there

File: mimecast/plugin.spec.yaml [hunk 5]
```diff
@@ -284,6 +284,12 @@ types:
       description: The action requested by the user
       type: string
       required: false
+  deletions:
```
Comment (jherlitz-r7) [inline mimecast/plugin.spec.yaml]: Is this custom type used anywhere?

**Comment by tylerslijboom-r7** [comment]
Screen Shot 2019-12-03 at 10 04 52 AM

File: mimecast/plugin.spec.yaml (hunk 3 above)
Comment (tylerslijboom-r7) [reply mimecast/plugin.spec.yaml]: this type was removed.

File: mimecast/komand_mimecast/actions/delete_managed_url/action.py (hunk 2 above)
Comment (tylerslijboom-r7) [reply mimecast/komand_mimecast/actions/delete_managed_url/action.py]: @lmilby-r7 This action will always return true from Mimecast, and I won't be able to confirm if the ID was successfully deleted.  This seems to be the only return type we can do.  It returns `True` explicitedly

Comment (jherlitz-r7) [reply mimecast/komand_mimecast/actions/delete_managed_url/action.py]: if that's true, I'm not sure having the output field of "success" is very helpful - might want to just omit it

Comment (lmilby-r7) [reply mimecast/komand_mimecast/actions/delete_managed_url/action.py]: @tylerslijboom-r7 okay, update bool(1) to True

File: mimecast/help.md [hunk 6]
```diff
@@ -230,19 +264,46 @@ This action is used to add an email address or domain to a group.
...
-  "email_address": "test10@example.com",
+  "email_address": "user@example.com",
   "internal": true
 }
 ```
 
+#### Delete Group Member
+
+This action is used to remove an email address or domain from a group.
+Delete on an email or domain that does not exist is will result in no operation performed.
```
Comment (gusingh-r7) [inline mimecast/help.md]: This sentence needs rework.

File: mimecast/plugin.spec.yaml [hunk 7]
```diff
@@ -601,6 +616,31 @@ actions:
...
+    description: Delete an email address or domain from a group
+    input:
+      id:
+        title: ID
+        description: The Mimecast ID of the group to remove from
+        type: string
+        required: true
+      email_address:
+        title: Email Address
+        description: The email address of to remove from group. Use either email address or domain
```
Comment (gusingh-r7) [inline mimecast/plugin.spec.yaml]:
```suggestion
        description: The email address to remove from group. Use either email address or domain
```

**Comment by tylerslijboom-r7** [comment]
> You need to regenerate the code as there are changes in plugin.spec.yaml.

I use icon-plugin generate help and i-p generate python --regenerate

**Pull request merged**